packages/plugin-twitter/src/actions/post.ts

```diff
@@ -123,45 +75,57 @@ export const postAction: Action = {
     name: "POST_TWEET",
     similes: ["TWEET", "POST", "SEND_TWEET"],
     description: "Post a tweet to Twitter",
-    validate: async (
-        runtime: IAgentRuntime,
-        message: Memory,
-        state?: State
-    ) => {
-        const hasCredentials =
-            !!process.env.TWITTER_USERNAME && !!process.env.TWITTER_PASSWORD;
-        elizaLogger.log(`Has credentials: ${hasCredentials}`);
-
-        return hasCredentials;
+    validate: async (runtime: IAgentRuntime) => {
+        const twitterClient = runtime.clients.twitter;
+        return !!twitterClient;
     },
     handler: async (
         runtime: IAgentRuntime,
-        message: Memory,
+        _message: Memory,
         state?: State
     ): Promise<boolean> => {
         try {
-            // Generate tweet content using context
-            const tweetContent = await composeTweet(runtime, message, state);
-
-            if (!tweetContent) {
-                elizaLogger.error("No content generated for tweet");
+            const context = composeContext({
+                state,
+                template: tweetTemplate,
+            });
+
+            // Generate tweet content
+            const newTweetContent = await generateText({
+                runtime,
+                context,
+                modelClass: ModelClass.SMALL,
+            });
+
+            // Clean up the content
+            let cleanedContent = newTweetContent
+                .replace(/^\s*{?\s*"text":\s*"|"\s*}?\s*$/g, "") // Remove JSON-like wrapper
+                .replace(/^['"](.*)['"]$/g, "$1") // Remove quotes
+                .replace(/\\"/g, '"') // Unescape quotes
+                .replace(/\\n/g, "\n\n") // Unescape newlines
+                .trim();
+
+            if (!cleanedContent) {
+                elizaLogger.error("No valid content generated for tweet");
                 return false;
             }
 
-            elizaLogger.log(`Generated tweet content: ${tweetContent}`);
+            // Handle truncation
+            const maxTweetLength = 280; // or get from config
```

Comment:
should consider whether the user has a premium account, as they might be able to post tweets longer than 280 characters. (Reference: https://github.com/elizaOS/eliza/pull/1339)

Reply:
yes i notice client-twitter uses different post types, i can use that

Reply:
Reference: https://github.com/elizaOS/eliza/blob/05463669844fc98f89151a1e914e4408b4d80f6b/packages/client-twitter/src/post.ts#L467